### PR TITLE
LFA-1628 Report confirmation statement as late when appropriate

### PR DESCRIPTION
**Pull request by mwestacott**

### CODE REVIEW

File: src/controllers/confirmation.controller.ts [hunk 1]
```diff
@@ -92,10 +92,21 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       return next(new Error("No new filing due date returned by the PTF API"));
     }
 
+    let overdueFiling: string;
+    if (companyProfile.isAccountsOverdue && !companyProfile.isConfirmationStatementOverdue) {
+      overdueFiling = "accounts";
+    } else if (companyProfile.isAccountsOverdue && companyProfile.isConfirmationStatementOverdue) {
+      overdueFiling = "accounts and confirmation statement";
+    } else {
+      // TODO Handle scenario when only confirmation statement is overdue LFA-1479
```
Comment (mwestacott) [inline src/controllers/confirmation.controller.ts:101]: Tackling this story next...

File: src/controllers/confirmation.controller.ts [hunk 2]
```diff
@@ -92,10 +92,21 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       return next(new Error("No new filing due date returned by the PTF API"));
     }
 
+    let overdueFiling: string;
+    if (companyProfile.isAccountsOverdue && !companyProfile.isConfirmationStatementOverdue) {
```
Comment (markpit) [inline src/controllers/confirmation.controller.ts:96]: how about:
`let overdueFiling: string;
    if (companyProfile.isAccountsOverdue) {
      overdueFiling = "accounts";
      if (companyProfile.isConfirmationStatementOverdue) {
        overdueFiling = overdueFiling.concat(" and confirmation statement");
      }
    } else {
      // TODO Handle scenario when only confirmation statement is overdue LFA-1479
      overdueFiling = "UNDEFINED";
    }`

File: src/controllers/confirmation.controller.ts [hunk 3]
```diff
@@ -92,10 +92,21 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       return next(new Error("No new filing due date returned by the PTF API"));
     }
 
+    let overdueFiling: string;
```
Comment (markpit) [inline src/controllers/confirmation.controller.ts:95]: how about:
let overdueFiling: string;
    if (companyProfile.isAccountsOverdue) {
      overdueFiling = "accounts";
      if (companyProfile.isConfirmationStatementOverdue) {
        overdueFiling = overdueFiling.concat(" and confirmation statement");
      }
    } else {
      // TODO Handle scenario when only confirmation statement is overdue LFA-1479
      overdueFiling = "UNDEFINED";
    }

File: src/controllers/confirmation.controller.ts (hunk 2 above)
Comment (markpit) [inline src/controllers/confirmation.controller.ts:96]:
```suggestion
    let overdueFiling: string;
    if (companyProfile.isAccountsOverdue) {
      overdueFiling = "accounts";
      if (companyProfile.isConfirmationStatementOverdue) {
        overdueFiling = overdueFiling.concat(" and confirmation statement");
      }
    } else {
      // TODO Handle scenario when only confirmation statement is overdue LFA-1479
      overdueFiling = "UNDEFINED";
    }
```

Comment (markpit) [reply src/controllers/confirmation.controller.ts:96]: what do you think?

Comment (markpit) [reply src/controllers/confirmation.controller.ts:96]: actually forget it, the conf statement bit would need another if, whereas yours doesn't

**Pull request merged**